core-api/src/main/java/com/sequenceiq/cloudbreak/validation/HttpContentSizeValidator.java

```diff
@@ -1,39 +1,47 @@
 package com.sequenceiq.cloudbreak.validation;
 
+import java.util.Optional;
+
+import javax.inject.Inject;
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
 import javax.ws.rs.core.Response.Status.Family;
 import javax.ws.rs.core.Response.StatusType;
 
+import org.apache.commons.io.FileUtils;
 import org.apache.commons.lang3.tuple.Pair;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
 
 import com.sequenceiq.cloudbreak.api.helper.HttpHelper;
 
+@Component
 public class HttpContentSizeValidator implements ConstraintValidator<ValidHttpContentSize, String> {
 
-    public static final int MAX_SIZE = 5;
-
-    public static final int MAX_IN_BYTES = 5 * 1024 * 1024;
-
     public static final String FAILED_TO_GET_BY_FAMILY_TYPE = "Failed to get response by the specified URL '%s' due to: '%s'!";
 
     public static final String INVALID_URL_MSG = "The value should be a valid URL and start with 'http(s)'!";
 
     public static final String FAILED_TO_GET_WITH_EXCEPTION = "Failed to get response by the specified URL!";
 
+    public static final String MESSAGE_VARIABLE = "contentLimit";
+
     private static final Logger LOGGER = LoggerFactory.getLogger(HttpContentSizeValidator.class);
 
-    private HttpHelper httpHelper = HttpHelper.getInstance();
+    @Inject
+    private Optional<ContentSizeProvider> contentSizeProvider;
 
-    @Override
-    public void initialize(ValidHttpContentSize constraintAnnotation) {
-    }
+    private HttpHelper httpHelper = HttpHelper.getInstance();
 
     @Override
     public boolean isValid(String value, ConstraintValidatorContext context) {
         LOGGER.info("content size validation was called. Value: {}", value);
+        if (contentSizeProvider.isEmpty()) {
+            throw new IllegalArgumentException("Missing object. Implement " + ContentSizeProvider.class.getName());
+        }
```

Comment:
I don't really like the idea that this would surface during runtime. Please ensure we have an implementation everywhere it's required